numba/tests/test_array_manipulation.py

```diff
@@ -52,146 +73,210 @@ def bad_float_index(arr):
 
+        self.assertIn("ravel", str(raises.exception))
+
+    def test_ravel_array_size_npm(self):
+        with self.assertRaises(errors.UntypedAttributeError) as raises:
+            self.test_ravel_array_size(flags=no_pyobj_flags)
+
+        self.assertIn("ravel", str(raises.exception))
+
     def test_transpose_array(self, flags=enable_pyobj_flags):
+        a = np.arange(9).reshape(3, 3)
+
         pyfunc = transpose_array
-        arraytype1 = types.Array(types.int32, 2, 'C')
-        arraytype2 = types.Array(types.int32, 2, 'C')
-        cr = compile_isolated(pyfunc, (arraytype1, arraytype2),
-                              flags=flags)
+        arraytype1 = typeof(a)
+        cr = compile_isolated(pyfunc, (arraytype1,), flags=flags)
         cfunc = cr.entry_point
 
-        a = np.arange(9).reshape(3, 3)
-        expected = np.arange(9).reshape(3, 3).transpose()
-        self.assertTrue(cfunc(a, expected))
+        expected = pyfunc(a)
+        got = cfunc(a)
+        np.testing.assert_equal(expected, got)
 
     def test_transpose_array_npm(self):
-        with self.assertTypingError():
-            self.test_transpose_array(flags=no_pyobj_flags)
+        self.test_transpose_array(flags=no_pyobj_flags)
 
     def test_squeeze_array(self, flags=enable_pyobj_flags):
+        a = np.arange(2 * 1 * 3 * 1 * 4).reshape(2, 1, 3, 1, 4)
+
         pyfunc = squeeze_array
-        arraytype1 = types.Array(types.int32, 2, 'C')
-        arraytype2 = types.Array(types.int32, 2, 'C')
-        cr = compile_isolated(pyfunc, (arraytype1, arraytype2),
-                              flags=flags)
+        arraytype1 = typeof(a)
+        cr = compile_isolated(pyfunc, (arraytype1,), flags=flags)
         cfunc = cr.entry_point
 
-        a = np.arange(2*1*3*1*4).reshape(2,1,3,1,4)
-        expected = np.arange(2*1*3*1*4).reshape(2,1,3,1,4).squeeze()
-        self.assertTrue(cfunc(a, expected))
+        expected = pyfunc(a)
+        got = cfunc(a)
+        np.testing.assert_equal(expected, got)
 
     def test_squeeze_array_npm(self):
-        with self.assertTypingError():
+        with self.assertRaises(errors.UntypedAttributeError) as raises:
             self.test_squeeze_array(flags=no_pyobj_flags)
 
-    def test_convert_array(self, flags=enable_pyobj_flags):
-        pyfunc = convert_array
-        arraytype1 = types.Array(types.int32, 1, 'C')
-        arraytype2 = types.Array(types.float32, 1, 'C')
-        cr = compile_isolated(pyfunc, (arraytype1, arraytype2),
-                              flags=flags)
+        self.assertIn("squeeze", str(raises.exception))
+
+    def test_convert_array_str(self, flags=enable_pyobj_flags):
+        a = np.arange(9, dtype='i4')
+
+        pyfunc = convert_array_str
+        arraytype1 = typeof(a)
+        cr = compile_isolated(pyfunc, (arraytype1,), flags=flags)
         cfunc = cr.entry_point
 
+        expected = pyfunc(a)
+        got = cfunc(a)
+        np.testing.assert_equal(expected, got)
```

Comment:
I don't think assert_equal() is enough here, it will only test the values. We should use assertPreciseEqual() to also test dtype, shape, etc.